aea/protocols/generator.py

```diff
@@ -919,216 +970,958 @@ def _message_class_str(self) -> str:
-        cls_str += "                ), \"Invalid 'target'. Expected 0 (because 'message_id' is 1). Found {}.\".format(\n"
-        cls_str += "                    self.target\n"
-        cls_str += "                )\n"
-        cls_str += "            else:\n"
-        cls_str += "                assert (\n"
-        cls_str += "                    0 < self.target < self.message_id\n"
-        cls_str += "                ), \"Invalid 'target'. Expected an integer between 1 and {} inclusive. Found {}.\".format(\n"
-        cls_str += "                    self.message_id - 1,\n"
-        cls_str += "                    self.target,\n"
-        cls_str += "                )\n"
-        cls_str += "        except (AssertionError, ValueError, KeyError) as e:\n"
-        cls_str += "            logger.error(str(e))\n"
-        cls_str += "            return False\n\n"
-        cls_str += "        return True\n"
+            _change_indent(-1)
+
+        cls_str += "\n"
+        cls_str += indent + "# Check correct content count\n"
+        cls_str += (
+            indent + "assert expected_nb_of_contents == actual_nb_of_contents, "
+            '"Incorrect number of contents. Expected {}. Found {}"'
+            ".format(expected_nb_of_contents, actual_nb_of_contents)\n\n"
+        )
+
+        cls_str += indent + "# Light Protocol Rule 3\n"
+        cls_str += indent + "if self.message_id == 1:\n"
+        _change_indent(1)
+        cls_str += (
+            indent
+            + "assert self.target == 0, \"Invalid 'target'. Expected 0 (because 'message_id' is 1). Found {}.\".format(self.target)\n"
+        )
+        _change_indent(-1)
+        cls_str += indent + "else:\n"
+        _change_indent(1)
+        cls_str += (
+            indent + "assert 0 < self.target < self.message_id, "
+            "\"Invalid 'target'. Expected an integer between 1 and {} inclusive. Found {}.\""
+            ".format(self.message_id - 1, self.target,)\n"
+        )
+        _change_indent(-2)
+        cls_str += indent + "except (AssertionError, ValueError, KeyError) as e:\n"
+        _change_indent(1)
+        cls_str += indent + "logger.error(str(e))\n"
+        cls_str += indent + "return False\n\n"
+        _change_indent(-1)
+        cls_str += indent + "return True\n"
+
+        return cls_str
+
+    def _end_state_enum_str(self) -> str:
+        """
+        Generate the end state Enum class.
+
+        :return: the end state Enum string
+        """
+        enum_str = indent + "class EndState(Enum):\n"
+        _change_indent(1)
+        enum_str += (
+            indent + '"""This class defines the end states of a dialogue."""\n\n'
+        )
+        enum_str += indent + "SUCCESSFUL = 0\n"
+        enum_str += indent + "DECLINED_CFP = 1\n"
+        enum_str += indent + "DECLINED_PROPOSE = 2\n"
+        enum_str += indent + "DECLINED_ACCEPT = 3\n"
+        _change_indent(-1)
+        return enum_str
```

Comment:
The end state should not be hard coded. This depends on the performatives.

Reply:
Done